### ISSUE TITLE
Rename graphdriver to storage driver in Experimental.md

### ISSUE THREAD

**Issue by tiborvass** [docker/docker#17996]
Title says it all

**Comment by HackToday** [comment]
@tiborvass from the doc said,

Docker graph driver plugins enable admins to use an external/out-of-process
graph driver for use with Docker engine. This is an alternative to using the
built-in storage drivers, such as aufs/overlay/devicemapper/btrfs.


It seems graph driver and storage driver is same. So not sure if needed to change graph to storage.

And docker code may have graph directory, I am not sure if it is good to just change doc.

**Comment by tiborvass** [comment]
Ping @cpuguy83

**Comment by thaJeztah** [comment]
I also looked at this, and if we want to rename this, we should probably rename the plugin API as well, given that `GraphDriver` is in those specs; https://github.com/docker/docker/blob/master/experimental/plugins_graphdriver.md#graphdriverremove

**Comment by cpuguy83** [comment]
wfm

**Comment by cpuguy83** [comment]
And there is container.GraphDriver in the inspect output.

**Comment by thaJeztah** [comment]
@cpuguy83 so if I interpret you "wfm" you want to have GraphDriver renamed to StorageDriver both in docs, code (and endpoints)?

Do we know if there are already people developing experimental GraphDriver plugins?

**Comment by cpuguy83** [comment]
On point 1, I'm neither + or - on it.

On point 2, yes there are a few people working on graphdriver plugins.
@gourao, @kolyshkin, @srust, possibly @erikh

**Comment by cpuguy83** [comment]
p.s., I already have this implemented with everything changed over to StorageDriver instead of GraphDriver but haven't pushed yet pending further discussion -- just so people don't waste their time doing it :) as well.

**Comment by cpuguy83** [comment]
Also @jlhawn mentioned interest.

**Comment by erikh** [comment]
I'm not working on one atm, but let me be a blowhard for a second: why isn't it called image driver instead? storage seems to generally conflict with the word 'volume' in my mind.

**Comment by gourao** [comment]
Yes, we are working on a graphdriver plugin based on Brian's experimental.  The goal of this driver is to provide a space efficient alternative that works in a multi node environment.

Regarding the name, I think StorageDriver is a bit misleading and can be confused with the Volume Driver.  I personally think Image Driver, or Graph Driver sound better.  My 2c.

**Comment by jlhawn** [comment]
"Image Driver" doesn't accurately describe what what the current "Graph Drivers" do. While storing of image rootfs snapshots a very important purpose, they are also used for storing the rootfs of containers and handling commits of containers to images. Some of the drivers can also be used as volume drivers (the "vfs" driver, for example, has been used as the default volume driver). I also think that the current name, "Graph Driver", is not descriptive enough. It sounds like it provides backend implementations for a graph database.

I think that "Rootfs Driver" would be the most accurate/descriptive name, given how we use it. It avoids conflating it with "Volume Driver" which many have already pointed out is also "storage".

TL;DR: I think `rootfs driver` would be a fitting name.

**Comment by erikh** [comment]
WFM, FWIW

**Comment by thaJeztah** [comment]
We should take into account that we use "Storage Driver" in the documentation and other places https://docs.docker.com/engine/userguide/storagedriver/imagesandcontainers/

(yes I understand the concerns of using "storage driver" as a name)

**Comment by tiborvass** [comment]
@gourao thank you for your input. Unfortunately, the storage drivers were already called graph drivers once. This issue is about being consistent with that decision, whether I agree with it or not.

**Comment by thaJeztah** [comment]
related https://github.com/docker/docker/pull/22228, and https://github.com/docker/docker/pull/23237

**Comment by thaJeztah** [comment]
There's not a lot of activity on this issue, and given that the code changes to rename it never made it in, I'm closing this issue, but feel free to ping me if you feel strongly about renaming it
